avatars/client_test.py

```diff
@@ -50,3 +39,44 @@ def test_url_is_rejected_if_it_contains_quotes(base_url: str) -> None:
     # env variable configuration.
     with pytest.raises(ValueError, match="not to contain quotes") as exc_info:
         api_client = ApiClient(base_url=base_url)
+
+
+import httpx
+
+RequestHandle = Callable[[httpx.Request], httpx.Response]
+
+
+def api_client_factory(handler: Optional[RequestHandle] = None) -> ApiClient:
+    """Generate an API client with a mock transport.
+
+    The handler returns an empty 200 response by default.
+    Consider overriding it with a custom handler for more complex tests.
+    """
+    if handler is None:
+        handler = lambda request: httpx.Response(200, json={})
+
+    transport = httpx.MockTransport(handler)
+    http_client = httpx.Client(base_url="http://localhost:8000", transport=transport)
+    return ApiClient(
+        base_url="http://localhost:8000",
```

Comment:
nit: base url could be a constant on the top of the file

Reply:
It's not needed, I could have written `"http://blabla"`, as we are not actually calling the API but using a `MockTransport`